content/blog/2018/12/2018-12-28-the-jenkins-features-that-i-use-like-and-not-necessarily-like.adoc

```diff
@@ -0,0 +1,45 @@
+
+image::https://github.com/alyssat/jenkins.io/blob/master/jenkins2018-12-blog-post.png[role="right"]
+      
+
+== In the above process, the following features of Jenkins or its plugins, in no particular order come-in extremely handy.
+
+* Upstream / Downstream jobs to implement the sequencing of the builds.
+* Organizing the 100s of jobs we have for the various active releases and branches we support at any time, into views and folders.
+* Using labels and multi-configuration project configurations to do the builds for a number of flavors of AP, WLC, and ME.
+* The Silent build feature of Gerrit plugin is useful for experimental builds on their way to become mandatory and also for builds whose return value is non-consequential, in which case the status is sent back in a different way.
+* Log parsing feature is of immense help when the console logs are humongous and also when we want the build to fail based on some error patterns in the logs.
+
+== Every tool comes with its own scope for improvement and here’s where I feel Jenkins needs some help:
+
+* Configuration is a one-way lane sometimes. It is not easy or possible to reconfigure. For instance, a freestyle project cannot be converted into a multi-configuration project later, when we want to add newer, but similar builds.
+* Need for plugins even for basic functionality, quite a bit of which can be made default in Jenkins.
+* Plugin security is a concern. There does not seem to be any certification process for the plug-ins, yet. There is a risk of using rogue plugins or carelessly written plugins to ruin the workflow.
+* Inefficient constructs in plugins go unchecked and can entice a novice to use them. For instance, plugins which provide busy polling the SCM system can cause substantial load.
+* Currently, Jenkins upgrade is non-trivial and can take days.
+* Some of the functionality is unreliable. For instance, CI builds sometimes do not start automatically.
+* Clumsy handling of humongous console logs. Browsers struggle to load the Full Console log. Need better ways of showing the logs.
+* On a minor note, the Execute Shell box does not have wrap-around, making it a challenge when using the trackpad on laptops.
+
+== Jenkins is still the best.
+
+Intuitive usage. In the decade-long use of Jenkins, I hardly had to refer to any documentation. The GUI is quite self-guiding.
+Plenty of Plugins (and functionality) to choose from.
+Simple concept of scheduling expanded beyond all recognition! A great example of tool innovation.
+Quite scalable, appropriate evolution to meet the growing needs of DevOps.
+Meets requirements for most CI / CD environments.
+
+
+As we saw the DevOps world - Jenkins World 2018 conference, is indeed centered around Jenkins.  Here is the link to the presentation I gave at the conference: https://sched.co/FCei
```

Comment:
I'm not sure what is indeed centered around Jenkins. Is there a word or phrase missing here?